test_scenarios/send_data.py

```diff
@@ -0,0 +1,185 @@
+        QUEUE_URL.format(AccountID=args.account_id,
+                         Region=args.region,
+                         QueueName=args.queue_name),
+                         data.get('Logs', []),
+                         data.get('LogType', ''),
+                         data.get('Format', 'json'))
+
+def process_file(event_time_shift, client, queue_url, logs, log_type, message_format):
+    # these 2 are special
+    if log_type == 'AWS.CloudTrail':
+        process_cloudtrail(event_time_shift, client, queue_url, logs, log_type)
+        return
+    if log_type == 'AWS.VPCFlow':
+        process_vpcflow(event_time_shift, client, queue_url, logs, log_type)
+        return
+
+    # nothing special to do
+    if message_format == 'json':
+        process_any_json(event_time_shift, client, queue_url, logs, log_type)
+        return
+    if message_format == 'raw':
+       process_any_raw(event_time_shift, client, queue_url, logs, log_type)
+       return
+
+    return
+
+def process_cloudtrail(event_time_shift, client, queue_url, logs, log_type):
+    logs = time_shift_json_logs(event_time_shift, logs, log_type)
+    logging.info('Sending [%d] CloudTrail logs...', len(logs))
+    # Wrap the CloudTrail in a 'Records' top-level key
+    resp = send_message(client, queue_url, {'Records': logs}, 'json')
+    logging.debug('Response: %s', resp['ResponseMetadata']['HTTPStatusCode'])
+
+def process_vpcflow(event_time_shift, client, queue_url, logs, log_type):
+    logs = time_shift_vpcflow_logs(event_time_shift, logs, log_type)
+    logging.debug('Sending VPC Flow log header')
+    resp = send_message(client, queue_url, FLOW_LOG_HEADER, 'raw')
+    logging.debug('Response: %s', resp['ResponseMetadata']['HTTPStatusCode'])
+
+    logging.info('Sending [%d] %s logs...', len(logs), log_type)
+    for indx, log in enumerate(logs):
+        resp = send_message(client, queue_url, log, 'raw')
+        logging.debug('Message [%d] response: %s', indx + 1, resp['ResponseMetadata']['HTTPStatusCode'])
+
+def process_any_json(event_time_shift, client, queue_url, logs, log_type):
+    logs = time_shift_json_logs(event_time_shift, logs, log_type)
+    logging.info('Sending [%d] %s logs...', len(logs), log_type)
+    for indx, log in enumerate(logs):
+        resp = send_message(client, queue_url, log, 'json')
+        logging.debug('Message [%d] response: %s', indx + 1, resp['ResponseMetadata']['HTTPStatusCode'])
+
+def process_any_raw(event_time_shift, client, queue_url, logs, log_type):
+    logs = time_shift_raw_logs(event_time_shift, logs, log_type)
+    logging.info('Sending [%d] %s logs...', len(logs), log_type)
+    for indx, log in enumerate(logs):
+        resp = send_message(client, queue_url, log, 'raw')
+        logging.debug('Message [%d] response: %s', indx + 1, resp['ResponseMetadata']['HTTPStatusCode'])
+
+# FIXME: we need to change to S3 files, sqs does not guarantee order of events and VPC flow needs a header to precede
```

Comment:
Could we use a FIFO SQS queue? Not sure if Panther supports this for SQS sources.

Reply:
We'd have to create the queues as FIFO in our integration

Reply:
Switched to S3 from SQS.